clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -10314,9 +10314,21 @@ void OffloadPackager::ConstructJob(Compilation &C, const JobAction &JA,
       File = C.getArgs().MakeArgString("@" + File);
 
     StringRef Arch;
+    std::string TransformedArch;
```

Comment:
Please, add a comment with the requirement for `TransformedArch` to outlive `Arch`. It's not obvious from the variable declaration.